controller/src/main/java/io/pravega/controller/task/Stream/StreamTransactionMetadataTasks.java

```diff
@@ -114,7 +128,7 @@ public Void initializeStreamWriters(final ClientFactory clientFactory,
 
     @VisibleForTesting
     public Void initializeStreamWriters(final String commitStreamName, final EventStreamWriter<CommitEvent> commitWriter,
-                                 final String abortStreamName, final EventStreamWriter<AbortEvent> abortWriter) {
+                                        final String abortStreamName, final EventStreamWriter<AbortEvent> abortWriter) {
```

Comment:
We introduced the `isReady` predicate, and initialization of stream-related properties after construction, to fix Issue #600 and it was fixed in PR #640. The reason for Issue #600 is that at bootstrap, creation of controller's streams, like commitStream, or abortStream may take a long time, especially if none of the segment stores have started. But we do not want the RPC or REST service to get blocked until these streams are created, and RPC and REST services require StreamTransactionTasks object. Hence we construct StreamTransactionTasks object at bootstrap and concurrently start (1) RPC service, (2) REST service, (3) Host-container mapper, (4) Creation of controller streams. Once the controller streams are created, we initialize the stream writers in StreamTransactionTasks.

RPC and REST services require StreamMetadataTasks and StreamTransactionTasks objects. The advantage of the existing bootstrap sequence is that although StreamTransactionTasks object may not be initialized initially, StreamMetadataTasks object is; and so, clients can successfully invoke methods like `createScope`, and `listScopes`, that are defined in StreamMetadataTasks object.

However, one could argue that the existing scheme is overzealous and we could postpone exposing RPC and REST services until all the internal components are successfully initialized. Filed an issue #1438 discuss and debate this alternative proposal.